Kudu.Core/Jobs/JobsManagerBase.cs

```diff
@@ -100,7 +102,10 @@ public void RegisterExtraEventHandlerForFileChange(Action<string> action)
 
         private void OnJobChanged(string jobName)
         {
-            JobListCache = null;
+            lock (jobsListCacheLockObj)
```

Comment:
Did some rebasing before pushing, can not send another update to this CR, closing and sending another one addressing David's comments